backend/service/chaos/experimentation/experimentstore/storer.go

```diff
@@ -180,25 +182,35 @@ func (s *storer) GetExperiments(ctx context.Context, configType string, status e
 
 	var experiments []*Experiment
 	for rows.Next() {
-		var details string
-		run := ExperimentRun{}
-		config := ExperimentConfig{Config: &any.Any{}}
+		var configId, details string
 		var terminationReason sql.NullString
-		err = rows.Scan(&run.Id, &run.StartTime, &run.EndTime, &run.CancellationTime, &run.CreationTime, &terminationReason, &config.Id, &details)
-		if terminationReason.Valid {
-			run.TerminationReason = terminationReason.String
-		}
+		run := ExperimentRun{}
 
+		err = rows.Scan(
+			&run.Id,
+			&run.StartTime,
+			&run.EndTime,
+			&run.CancellationTime,
+			&run.CreationTime,
+			&terminationReason,
+			&configId,
+			&details)
 		if err != nil {
 			return nil, err
 		}
 
-		err = protojson.Unmarshal([]byte(details), config.Config)
+		config, err := NewExperimentConfig(configId, details)
 		if err != nil {
-			return nil, err
+			s.logger.Errorw("failed to initialize experiment config", "configId", configId, "runId", run.Id)
+			s.configInitializationErrorCount.Inc(1)
```

Comment:
initialize sounds a bit strange since we're reading a value from the database, maybe deserialization ?

Reply:
fixed